shared/src/balancer_event_handler.rs

```diff
@@ -129,19 +345,20 @@ impl EventStoring<ContractEvent> for BalancerPools {
             balancer_events.len(),
             range.start().to_u64()
         );
-        todo!()
+        // Not sure we will event need to replace events as we don't store the events themselves.
```

Comment:
I think this kind of question shouldn't go in the code but on the PR if possible (it would be weird for future maintainers to read this).

To answer this question specifically, I think technically we need to replace events in case of reorgs (otherwise we might store a pool which was only mined in the uncle chain). Unlikely, but given that the logic is quite straight forward, I'd leave it in.